rlog/2023-08-04-wakurtosis-waku-scallability.mdx

```diff
@@ -0,0 +1,99 @@
+However, while it offers granular insights, process-level monitoring can be more resource-intensive and might not always capture overarching system behavior.
+
+### Performance Metrics
+Data sources like Cadvisor/Prometheus, Docker Stats, and process-level monitoring enhance the reliability of Wakurtosis, offering a comprehensive view of system performance.
+
+- Hardware Level Metrics: Emphasis on memory usage, CPU consumption, Disk I/O, and Network I/O.
+- Topology Level Metrics: Focuses on the efficiency of message propagation across the network, including metrics like message injection, propagation times, and message loss. 
+
+### Scalability
+To overcome scalability challenges, Wakurtosis employs a multi-node approach, running several nodes within one container. 
+This method supports simulations with over 1,000 nodes on a single machine. 
+However, this can introduce unforeseen network effects, potentially affecting some metrics. 
+Nevertheless, Wakurtosis offers the flexibility to choose between this and a 1-to-1 simulation, catering to the specific needs of each test scenario.
+
+## Examining the Waku Protocol
+
+### Simulation Setup
+To evaluate Waku under varied conditions, we conducted simulations across a range of network sizes, topologies, and message rates. 
+Each simulation lasted 3 hours to allow the network to stabilize. The network sizes explored included 75, 150, 300, and 600 nodes. 
+For non-Discv5 simulations, we used static topologies with average node degrees of *K=3*, *K=13*, and *K=50*. 
+For Discv5 simulations, we configured max_peers settings of 12 and 50 to approximate similar average degrees. 
+To stress test message throughput, we simulated message rates of 1, 10, and 100 messages per second. 
+This allowed us to observe performance under low, moderate, and high traffic loads. We leveraged Wakurtosis' ability to run multi-node simulations to achieve networks at the scale of 600 nodes. 
+This combination of network sizes, topologies, message rates, and hardware configurations enabled us to comprehensively evaluate Waku's performance and scalability boundaries under diverse conditions.
+
+### Results
+Despite certain scalability challenges, Wakurtosis has proved effective in simulating Waku with over 1,000 nodes on a single machine. 
+We examined the behaviours of the Waku protocol with multiple network sizes, message rates and peer discovery strategies. 
+Our analysis under baseline conditions --- static network with average degrees of *K=3* and *K=13* --- revealed promising scalability and stability from a memory standpoint. 
+Still, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
```

Comment:
"bandwidth usage improved with an increase..."

I think this is simply an effect of hardware constraints, so perhaps need to be clarified or left out. Just in case people believe we're capable of miracles. :)

Reply:
LOL ... indeed!